pkg/auth/cli.go

```diff
@@ -45,7 +47,8 @@ type LogoutOptions struct {
 // GetLoginFlags defines and returns login flags for containers tools
 func GetLoginFlags(flags *LoginOptions) *pflag.FlagSet {
 	fs := pflag.FlagSet{}
-	fs.StringVar(&flags.AuthFile, "authfile", GetDefaultAuthFile(), "path of the authentication file. Use REGISTRY_AUTH_FILE environment variable to override")
+	fs.StringVar(&flags.AuthFile, "authfile", "", "path of the authentication file. Use REGISTRY_AUTH_FILE environment variable to override")
+	fs.StringVar(&flags.DockerCompatAuthFile, "compat-auth-file", "", "path of a Docker-compatible config file to update instead")
```

Comment:
Should this be named `compat-authfile`? Ii is inconsistent to have `authfile` as one word and then another option split it as `auth-file`

Reply:
My very weak preference is to keep the dash, and to think of the previous `authfile` as a mistake.

The spelling with the dash was also suggested by someone else in the design doc, and widely approved there — though I’m not 100% clear all of the reviewers were focused on the dash question.

Reply:
Yeah I can live with either one as well, I just noticed the inconsistency when looking at this here.
If people didn't complain in the design doc then I am fine keeping it like this.

Reply:
can we rename `authfile` to `auth-file` and add an alias to maintain `authfile`?  Otherwise, I find it confusing too that one has the dash and one hasn't

Reply:
I am OK with the plan of migrating after 4.8 given our current timing issues

Reply:
I have filed https://github.com/containers/podman/issues/20693 to make sure the `--auth-file` renaming does not get lost.